ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -71,186 +73,201 @@ export interface Props {
-      this.state = {
-        loading: true,
-        error: false,
-        runtime: null
-      };
-    }
+  const activeRuntimeOp: RuntimeOperation = runtimeOps.opsByWorkspaceNamespace[workspace.namespace];
+  const {status = RuntimeStatus.Unknown, toolDockerImage = null, dataprocConfig = {}, gceConfig = {}} = currentRuntime || {};
+  const masterMachineType = !!dataprocConfig ? dataprocConfig.masterMachineType : gceConfig.machineType;
+  const masterDiskSize = !!dataprocConfig ? dataprocConfig.masterDiskSize : gceConfig.bootDiskSize;
+  const selectedMachineType = selectedMachine && selectedMachine.name;
 
-    async componentDidMount() {
-      // TODO(RW-5420): Centralize a runtimeStore.
-      let runtime = null;
-      let error = false;
+  useOnMount(() => {
+    const aborter = new AbortController();
+    const {namespace} = workspace;
+    const loadRuntime = async() => {
+        // TODO(RW-5420): Centralize a runtimeStore.
       try {
-        const promise = runtimeApi().getRuntime(this.props.workspace.namespace, {signal: this.aborter.signal});
-        updateRuntimeOpsStoreForWorkspaceNamespace(this.props.workspace.namespace, {
+        const promise = runtimeApi().getRuntime(namespace, {signal: aborter.signal});
+        updateRuntimeOpsStoreForWorkspaceNamespace(namespace, {
           promise: promise,
           operation: 'get',
-          aborter: this.aborter
+          aborter: aborter
         });
-        runtime = await promise;
       } catch (e) {
         // 404 is expected if the runtime doesn't exist, represent this as a null
         // runtime rather than an error mode.
         if (e.status !== 404) {
-          error = true;
+          setError(true);
         }
+      } finally {
+        setLoading(false);
       }
-      markRuntimeOperationCompleteForWorkspace(this.props.workspace.namespace);
-      this.setState({
-        runtime,
-        error,
-        loading: false
-      });
-    }
-
-    render() {
-      const {runtimeOps, workspace} = this.props;
-      const {loading, error, runtime} = this.state;
+      markRuntimeOperationCompleteForWorkspace(namespace);
+    };
 
-      const activeRuntimeOp: RuntimeOperation = runtimeOps.opsByWorkspaceNamespace[workspace.namespace];
-
-      if (loading) {
-        return <Spinner style={{width: '100%', marginTop: '5rem'}}/>;
-      } else if (error) {
-        return <div>Error loading compute configuration</div>;
-      } else if (!runtime) {
-        // TODO(RW-5591): Create runtime page goes here.
-        return <React.Fragment>
-          <div>No runtime exists yet</div>
-          {activeRuntimeOp && <hr/>}
-          {activeRuntimeOp && <div>
-            <ActiveRuntimeOp operation={activeRuntimeOp.operation} workspaceNamespace={workspace.namespace}/>
-          </div>}
-        </React.Fragment>;
-      }
+    loadRuntime();
+    return () => aborter.abort();
+  });
 
-      const isDataproc = !!runtime.dataprocConfig;
+  if (loading) {
```

Comment:
If you're using the `CurrentRuntimeStore` to get information about the runtime, then `loading`, `error`, etc. state should become unnecessary; instead you'd want to look at whether the runtime store is null / undefined